pkg/util/net.go

```diff
@@ -3,22 +3,27 @@ package util
 import (
 	"fmt"
 	"net"
+
+	"github.com/go-kit/kit/log/level"
 )
 
 // GetFirstAddressOf returns the first IPv4 address of the supplied interface names.
 func GetFirstAddressOf(names []string) (string, error) {
 	for _, name := range names {
 		inf, err := net.InterfaceByName(name)
 		if err != nil {
+			level.Error(Logger).Log("msg", "error getting interface", "inf", name, "err", err)
 			continue
 		}
 
 		addrs, err := inf.Addrs()
 		if err != nil {
-			return "", err
+			level.Error(Logger).Log("msg", "error getting addresses for interface", "inf", name, "err", err)
```

Comment:
Seems more like a warning than an error, given we continue.  If there are none found a real error will be returned.